src/components/Popups/ReplyField/ReplyField.tsx

```diff
@@ -50,6 +90,69 @@ export default class ReplyField extends React.Component<Props, State> {
         this.saveCursorPosition();
     }
 
+    /**
+     * Copied from BUIE/components/form-elements/draft-js-mention-selector/DraftJSMentionSelectorCore.js
+     */
+    getActiveMentionForEditorState(editorState: EditorState): Mention | null {
+        const { mentionPattern } = this.state;
+
+        const contentState = editorState.getCurrentContent();
+        const selectionState = editorState.getSelection();
+
+        const startKey = selectionState.getStartKey();
+        const activeBlock = contentState.getBlockForKey(startKey);
+
+        const cursorPosition = selectionState.getStartOffset();
+
+        let result = null;
+
+        // Break the active block into entity ranges.
+        activeBlock.findEntityRanges(
+            character => character.getEntity() === null,
+            (start, end) => {
+                // Find the active range (is the cursor inside this range?)
+                if (start <= cursorPosition && cursorPosition <= end) {
+                    // Determine if the active range contains a mention.
+                    const activeRangeText = activeBlock.getText().substr(start, cursorPosition - start);
+                    const mentionMatch = activeRangeText.match(mentionPattern);
+
+                    if (mentionMatch) {
+                        result = {
+                            blockID: startKey,
+                            mentionString: mentionMatch[2],
+                            mentionTrigger: mentionMatch[1],
+                            start: start + (mentionMatch.index as number),
+                            end: cursorPosition,
+                        };
+                    }
+                }
+
+                return null;
+            },
+        );
+
+        return result;
+    }
+
+    getContainerRect = (): DOMRect | null =>
+        document
+            .getElementsByClassName('ba-ReplyField')[0]
+            .getClientRects()
+            .item(0);
+
+    updateRect = (): void => {
+        const { containerRect: prevRect } = this.state;
+        const rect = this.getContainerRect();
+        const isSame = prevRect?.x === rect?.x && prevRect?.y === rect?.y;
+
+        if (!isSame) {
+            this.showPopup();
```

Comment:
This method is doing more than its name (`updateRect`) would imply. Is it doing too much?